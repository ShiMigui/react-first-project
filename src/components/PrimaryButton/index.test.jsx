import { render, screen } from '@testing-library/react';
import { PrimaryButton } from '.';
import userEvent from '@testing-library/user-event';

const mock = {
    text: 'Button',
    onClick: jest.fn()
};

const getEl = (disabled, props = mock) => <PrimaryButton {...props} disabled={disabled} />;

describe('<PrimaryButton />', () => {
    it('must be in the document', () => {
        render(getEl());

        const btn = screen.getByRole('button', { name: mock.text });
        expect(btn).toBeInTheDocument();
    })

    it('must be disabled', () => {
        render(getEl(true));

        const btn = screen.getByRole('button', { name: mock.text });
        expect(btn).toBeDisabled();
    })

    it('must call function OnClick()', () => {
        render(getEl(false));

        const btn = screen.getByRole('button', { name: mock.text });
        userEvent.click(btn);

        expect(btn).toBeEnabled();
        expect(mock.onClick).toHaveBeenCalledTimes(1);
    })

    it('must match with snapshots', () => {
        mock.text = 'Button enabled';
        render(getEl(false));

        mock.text = 'Button disabled';
        render(getEl(true, mock));

        const disabled = screen.getByRole('button', { name: /disabled/i });
        const enabled = screen.getByRole('button', { name: /enabled/i });

        expect(disabled).toMatchSnapshot();
        expect(enabled).toMatchSnapshot();
    })
})