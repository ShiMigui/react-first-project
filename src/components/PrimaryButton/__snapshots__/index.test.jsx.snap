// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PrimaryButton /> must match with snapshots 1`] = `
<button
  class="primary-button"
  disabled=""
>
  Button disabled
</button>
`;

exports[`<PrimaryButton /> must match with snapshots 2`] = `
<button
  class="primary-button"
>
  Button enabled
</button>
`;
